nlp/voice/backend-microsoft.js

```diff
@@ -0,0 +1,143 @@
+// -*- mode: js; indent-tabs-mode: nil; js-basic-offset: 4 -*-
+//
+// This file is part of Almond
+//
+// Copyright 2019-2020 The Board of Trustees of the Leland Stanford Junior University
+//
+// Author: Euirim Choi <euirim@cs.stanford.edu>
+//
+// See COPYING for details
+"use strict";
+
+const Tp = require('thingpedia');
+const fs = require('fs');
+const xmlbuilder = require('xmlbuilder');
+const https = require('https');
+const {
+  AudioInputStream,
+  ResultReason,
+  AudioConfig,
+  SpeechConfig,
+  SpeechRecognizer,
+} = require('@euirim/microsoft-cognitiveservices-speech-sdk');
```

Comment:
Looks like they finally applied my PR fixing the typescript bug and uploaded the change to NPM. We may consider moving away from this fork in the near future.